wordpress-mobile/WordPress-Android
# Stats Traffic Charts

https://github.com/wordpress-mobile/jetpack-issue-repo/issues/14#issue-2147025725

Adds bar graphs for Week, Month, and Year granularity

| Day | Week | Month | Year |
|--------|--------|--------|--------|
| ![Screenshot_20240220_145219](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/ee5079e8-9768-4782-8568-c8dca40b564f) | ![Screenshot_20240223_163324](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/c2414551-d50a-47b8-ae3a-5f787b839f14) |![Screenshot_20240223_163355](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/f7b93015-d209-46b4-b7cf-7f10e527c148) | ![Screenshot_20240223_163417](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/59af2554-931d-4cb7-9ee6-8ed9eb0d32cb) | 

-----

## To Test:

Test 1:
- Enable `stats_traffic_tab` feature flag (Me -> Debug settings - Remote Features)
- Go to Stats
- `Verify` only Traffic tab, and Insights tabs are shown as in after column 
- `Switch` to Traffic tab if not on it already
- `Select` By day from the drop-down
- `Verify` that Today card is shown in the image above
- `Select` By week from the drop-down
- `Verify` a Bar chart is shown with seven bars for 7 days in a week

> [!NOTE] 
> This is still work in progress.  Date selector, and graphs need to be in sync
> You may also check `By month`, and `By year` options from the drop-down

Test 2:

- Disable feature flag
- `Verify` Stats tabs all show up, and work as before



-----

## Regression Notes

1. Potential unintended areas of impact

    - Behind feature flag

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Tested 

3. What automated tests I added (or what prevented me from doing so)

    - Updated existing unit tests

-----

## PR Submission Checklist:


-----

## Testing Checklist:
